swift/Workflow/Sources/SubtreeManager.swift

```diff
@@ -49,30 +48,32 @@ extension WorkflowNode {
             /// as a result of this call to `render`.
             self.childWorkflows = context.usedChildWorkflows
             self.childWorkers = context.usedChildWorkers
+            /// Merge all of the signals together from the subscriptions.
+            self.subscriptions = Subscriptions(eventSources: context.eventSources, actionPipe: ActionPipe())
 
-            /// Finally, merge all of the events together from the new set of children and start listening to them
-            for child in self.childWorkflows {
-                child.value.onUpdate = { [weak self] output in
-                    self?.handle(output: output)
-                }
-            }
+            /// Capture all the pipes to be enabled after render completes.
+            self.actionPipes = context.actionPipes
+            self.actionPipes.append(self.subscriptions.actionPipe)
 
-            for worker in self.childWorkers {
-                worker.onUpdate = { [weak self] output in
+            /// Return the rendered result
+            return rendering
+        }
+
+        /// Enable the actionPipes for the previous rendering. The actionPipes are not valid until this has
+        /// be called. If is an error to call this twice without generating a new rendering.
+        func enableRendering() {
```

Comment:
This is now the second pass on a render loop. All event sources go through the `ActionPipe`s, so it's a singular path that is synchronous to validate we are on the correct dispatch queue, as well as having been enabled.